zebra-rpc/src/methods.rs

```diff
@@ -69,38 +73,83 @@ pub trait Rpc {
         &self,
         raw_transaction_hex: String,
     ) -> BoxFuture<Result<SentTransactionHash>>;
+
+    /// getblock
+    ///
+    /// Returns requested block by height, encoded as hex.
+    ///
+    /// zcashd reference: <https://zcash.github.io/rpc/getblock.html>
+    ///
+    /// Result:
+    /// {
+    ///      "data": String, // The block encoded as hex
+    /// }
+    ///
+    /// Note 1: We only expose the `data` field as lightwalletd uses the non-verbose
+    /// mode for all getblock calls: <https://github.com/zcash/lightwalletd/blob/v0.4.9/common/common.go#L232>
+    ///
+    /// Note 2: `lightwalletd` only requests blocks by height, so we don't support
+    /// getting blocks by hash.
+    ///
+    /// Note 3: The `verbosity` parameter is ignored but required in the call.
+    #[rpc(name = "getblock")]
+    fn get_block(&self, height: Height, verbosity: u8) -> BoxFuture<Result<GetBlock>>;
 }
 
 /// RPC method implementations.
-pub struct RpcImpl<Mempool>
+
+pub struct RpcImpl<Mempool, State>
 where
-    Mempool: tower::Service<mempool::Request, Response = mempool::Response, Error = BoxError>,
+    Mempool: Service<mempool::Request, Response = mempool::Response, Error = BoxError>,
+    State: Service<
+        zebra_state::Request,
+        Response = zebra_state::Response,
+        Error = zebra_state::BoxError,
+    >,
 {
     /// Zebra's application version.
-    app_version: String,
-
+    pub app_version: String,
```

Comment:
Minor: we can keep these fields private because they don't need to be accessed externally.

Reply:
Fixed this at https://github.com/ZcashFoundation/zebra/pull/3707/commits/6a4055f2030fb896b89f9c015edb6aa0c97be606